kernelci/dashboard
# Feat/add filters tests

# Add filter to tests and boots in tree details


- Changed filter lists to sets for better performance and clarity.
- Updated variable names for better readability.
- Added additional filters for configs, compiler, and architecture.
- Refactored the filtering logic in the `TreeDetailsSlow` view.
- Updated the frontend to handle new filter parameters.
- Removed unnecessary useSearch hook in BootsTab and TestsTab.


## How to test
Go to any tree that has tests eg: (broonie-sounds), and mess with the filters, see if it affects the boots/tests tab 

It is really hard to know if it is right, so, check the test details to see which filters you can use, because not all builds have tests

Coses #302, #326
